packages/flutter_tools/test/commands.shard/hermetic/update_packages_test.dart

```diff
@@ -173,6 +174,28 @@ void main() {
       ),
     });
   });
+
+  group('generateFakePubspec', () {
+    testUsingContext('constrains package versions to >= previous version if doUpgrade: true', () {
```

Comment:
Is there already a test that covers the `doUpgrade = false` case?

Reply:
Added another test case for doUpgrade = false